GoogleSignIn/Sources/GIDGoogleUser_Private.h

```diff
@@ -31,17 +31,16 @@ NS_ASSUME_NONNULL_BEGIN
 /// A completion block that takes a `GIDGoogleUser` or an error if the attempt to refresh tokens was unsuccessful.
 typedef void (^GIDGoogleUserCompletion)(GIDGoogleUser *_Nullable user, NSError *_Nullable error);
 
-// Internal methods for the class that are not part of the public API.
-@interface GIDGoogleUser () <GTMAppAuthFetcherAuthorizationTokenRefreshDelegate,
-                             OIDAuthStateChangeDelegate>
+/// Internal methods for the class that are not part of the public API.
+@interface GIDGoogleUser () <GTMAuthSessionDelegate, OIDAuthStateChangeDelegate>
```

Comment:
Removed the declare protocol conformance. It was a vestige of past code.

The relationship is similar to [what was previously in place](https://github.com/google/GoogleSignIn-iOS/blob/main/GoogleSignIn/Sources/GIDGoogleUser.m#L232-L234). Previously, if on iOS and if EMM was supported, we would create an instance of a `GTMAppAuthFetcherAuthorization` [subclass within GoogleSignIn-iOS](https://github.com/google/GoogleSignIn-iOS/blob/main/GoogleSignIn/Sources/GIDAppAuthFetcherAuthorizationWithEMMSupport.m) that supported EMM error handling (with the `OIDAuthState` at hand). Thus, the `GIDGoogleUser` could potentially expose a [`fetcherAuthorizer`](https://github.com/google/GoogleSignIn-iOS/blob/main/GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h#L71) that would capture the fetcher's callbacks and provide GSI with a chance to substitute a different (EMM) error if one occurs.

In this change, we got rid of the subclass because GTMAppAuth is now written fully in Swift, and `GTMAppAuthFetcherAuthorization` has been replaced with `GTMAuthSession`. That means `GTMAuthSession` cannot be subclassed in Objective-C. The solution we came up with was that GTMAppAuth would expose a new protocol [`GTMAuthSessionDelegate`](https://github.com/google/GTMAppAuth/blob/3.0.0/GTMAppAuth/Sources/AuthSessionDelegate.swift) that provided the various callbacks `GTMAppAuthFetcherAuthorization` previously provided, and also added an interface for EMM error handling. Since `GIDEmmSupport` already existed and did the error handling for this case, I just added the `GTMAuthSessionDelegate` conformance there.

Internally, [`GTMAuthSession` will check to see if it has a delegate](https://github.com/google/GTMAppAuth/blob/3.0.0/GTMAppAuth/Sources/AuthSession.swift#L290) and will ask it for an updated error when it is authorizing a request.

Putting it all together, if clients wish to use `GIDGoogleUser`'s `fetcherAuthorizer` to authorize their fetches, then they will get EMM error handling support.

Let me know if this makes sense!

Reply:
Thanks for the explanation. It's not as straightforward as it used to be now that `GIDEmmSupport` conforms to `GTMAuthSessionDelegate`, and `GIDAuthSession` conforms to `GTMAppAuthFetcherAuthorization`,  but I think this is a reasonable path moving forward. Resolving this conversation.